src/dotnet/APIView/APIViewWeb/Pages/Assemblies/Conversation.cshtml.cs

```diff
@@ -86,5 +96,90 @@ public async Task<IActionResult> OnGetAsync(string id)
             }
             return threadDict.OrderByDescending(kvp => Review.Revisions.IndexOf(kvp.Key));
         }
+
+        private Dictionary<UsageSampleRevisionModel, List<CommentThreadModel>> ParseUsageSampleThreads(IEnumerable<CommentThreadModel> threads)
+        {
+            var threadDict = new Dictionary<UsageSampleRevisionModel, List<CommentThreadModel>>();
+            
+            if (Sample == null)
+            {
+                return threadDict;
+            }
+
+            foreach (var thread in threads)
+            {
+                foreach (var comment in thread.Comments)
+                {
+                    if (comment.IsUsageSampleComment)
+                    {
+                        var sampleRevision = Sample.Revisions.Where(e => e.FileId.Equals(comment.ElementId.Split("-").First())).First();
+                        if (sampleRevision.RevisionIsDeleted)
+                        {
+                            continue;
+                        }
+
+                        if(!threadDict.ContainsKey(sampleRevision))
+                        {
+                            threadDict.Add(sampleRevision, new List<CommentThreadModel>());
+                        }
+                        threadDict[sampleRevision].Add(thread);
+
+                    }
+                }
+            }
+
+            return threadDict;
+        }
+
+        private async Task<IEnumerable<List<string>>> getUsageSampleLines()
+        {
+            List<List<string>> lines = new List<List<string>>();
+
+            foreach (var revision in Sample.Revisions.ToArray().Reverse())
+            {
+                if (revision.RevisionIsDeleted)
+                {
+                    lines.Add(new List<string>());
+                    if (revision.RevisionNumber == 1)
+                    {
+                        break;
+                    }
+                    continue;
+                }
+
+                string rawContent = await _samplesManager.GetUsageSampleContentAsync(revision.FileId);
+                var sampleLines = new List<string>();
+
+                rawContent = rawContent.Trim();
```

Comment:
Can `rawContent` be null (returned by GetUsageSampleContentAsync)?

Reply:
Yes, I'll add a check for this